mrmustard/lab_dev/states/ket.py

```diff
@@ -113,12 +113,16 @@ def from_ansatz(
         ansatz: PolyExpAnsatz | ArrayAnsatz | None = None,
         name: str | None = None,
     ) -> State:
+        if not isinstance(modes, set) and sorted(modes) != list(modes):
+            raise ValueError(f"Modes must be sorted. Got {modes}")
         modes = set(modes)
         if ansatz and ansatz.num_vars != len(modes):
             raise ValueError(
                 f"Expected an ansatz with {len(modes)} variables, found {ansatz.num_vars}."
             )
         wires = Wires(modes_out_ket=modes)
+        if isinstance(ansatz, ArrayAnsatz):
+            wires = wires.to_fock(ansatz.array.shape)
```

Comment:
Not sure how I feel about the idea of passing information related to the ansatz (i.e fock shape) into the wires. What is the motivation for it?

Reply:
It was so that when you need to change representation you have the arguments for BtoFoo.inverse()